tests/unit/EntityId/PrefixingEntityIdParserTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\BasicEntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixingEntityIdParser;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\SuffixEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefixingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @dataProvider validInputProvider
+	 */
+	public function testParse( $prefix, $input, $expected ) {
+		$parser = new PrefixingEntityIdParser( $prefix, new BasicEntityIdParser() );
+		$this->assertEquals( $expected, $parser->parse( $input ) );
+	}
+
+	public function validInputProvider() {
+		return array(
+			array( 'wikidata:', 'Q1337', new ItemId( 'wikidata:Q1337' ) ),
+			array( 'foo:', 'P123', new PropertyId( 'foo:P123' ) ),
+			array( 'foo:', 'bar:Q1337', new ItemId( 'foo:bar:Q1337' ) ),
```

Comment:
@thiemowmde I believe questions you ask are a bit beyond the scope of what this "parser" (which might not end up being call a parser any more) is supposed to do.
Your questions rather seem to me to be about the idea of "foreign entity ids" in general. It is definitely good that you raised them.
I don't know if I am the best person to try to answer them, and if you want to discuss them here, but to pick few of them:
> How can code ever assume a prefixed EntityId like "enbase:Q1" is not prefixed, and prefix it with "debase:"?

Given the current (I believe unwritten) specification of foreign entity IDs, the ID could contain multiple parts separated by a colon. Left-most of them is the "foreign repository name", while all other is the ID coming from that foreign repo. In order to resolve the ID (ie. to fetch the entity the ID is referring to) there will be some lookup mechanism which will talk to the foreign repo.
So if I had a local Wikibase instance, and two foreign repos configured, using names "enbase" and "debase", when I see an entity ID like "enbase:Q1" or "enbase:debase:Q1" what I would know is that these entities are coming from the repo I call "enbase". I don't really need to know what "Q1" and "debase:Q1" stand for in this case. When resolving "enbase:debase:Q1" requires enbase repo to do lookup in another foreign repo (which enbase labelled as "debase") then fine, this will happen. My local repo will not really know about that.

So basically, as @JeroenDeDauw already said while I was typing, this "parser" does not know specifics of implementation of "foreign ids", this is done in other places.

Checking if the prefix is already in place would avoid having things like "enbase:enbase:Q1" but you're right when multiple prefixes are allowed, some "loops" on further "positions" might be possible, and something should/could be done about them. Still I don't think it is to be concern of this simple prefixing "parser".

Also, BTW, I don't know what you mean by "localized" in
>  Let's say I do have two localized Wikibase repositories, "debase" and "enbase". 

How Wikibase instance could be localized?